src/ui/ImageUploadEditor.tsx

```diff
@@ -44,28 +54,65 @@ class ImageUploadEditor extends React.Component<any, any> {
                     <fieldset>
                         <legend>Upload Image</legend>
                         <div className={prefixed('image-upload-editor-body')}>
-                            <div className={prefixed('image-upload-editor-label')}>
-                                {label}
-                            </div>
-                            <input
-                                accept="image/png,image/gif,image/jpeg,image/jpg"
-                                className={prefixed('image-upload-editor-input')}
-                                disabled={pending}
-                                id={id}
-                                key={id}
-                                onChange={this._onSelectFile}
-                                type="file"
-                            />
+                            {this._renderImage(label)}
                         </div>
+                        <input
+                            className={prefixed('image-upload-editor-alt-text-input')}
+                            onChange={this._onAltTextChange}
+                            placeholder="Alternative Text"
+                            type="text"
+                            value={alt || ''}
+                        />
                     </fieldset>
                     <div className={prefixed('form-buttons')}>
                         <CustomButton label="Cancel" onClick={this._cancel} />
+                        <CustomButton
+                            active={!!image}
+                            disabled={!image}
+                            label="Insert"
+                            onClick={this._insert}
+                        />
                     </div>
                 </form>
             </div>
         )
     }
 
+    _renderImage = (label) => {
+        const {image, id, pending, alt} = this.state
+
+        if(!image || (image && !image.src)) {
+            return (
+                <>
+                    <div className={prefixed('image-upload-editor-label')}>
+                        {label}
+                    </div>
+                    <input
+                        accept="image/png,image/gif,image/jpeg,image/jpg"
+                        className={prefixed('image-upload-editor-input')}
+                        disabled={pending}
+                        id={id}
+                        key={id}
+                        onChange={this._onSelectFile}
+                        type="file"
+                    />
+                </>
+            )
+        }
+        return (
+            <img alt={alt} height="100%" src={image.src}/>
+        )
+    }
+
+    _onAltTextChange = (e: React.SyntheticEvent<HTMLInputElement>) => {
+        // @ts-ignore
+        const alt = e.target.value
```

Comment:
I now think I see why you needed the `ts-ignore`, try changing the type to be `ChangeEvent`:
```
onAltTextChange = (e: React.ChangeEvent <HTMLInputElement>) => {
...
}
```

Reply:
yes, that was it.